src/fu-engine.c

```diff
@@ -1054,17 +1044,83 @@ fu_engine_check_hardware_requirement (FuEngine *self, AsApp *app, GError **error
 }
 #endif
 
+static GHashTable *fu_engine_get_report_metadata (FuEngine *self);
+
+static gchar *
+fu_engine_requirement_convert_key (const gchar *name)
+{
+	struct {
+		const gchar *old;
+		const gchar *new;
+	} project_names[] = {
+		{ "FwupdVersion",		"org.freedesktop.fwupd" },
+		{ "FwupdateVersion",		"com.redhat.fwupdate" }, //FIXME?
+		{ "EfivarVersion",		"com.redhat.efivar" }, //FIXME?
```

Comment:
I think it would be better to let the plugins report these requirements names through a vfunc than hardcode them in `fu-engine`.

Reply:
I know what you mean, it's clumsy and inelegant. I'll have a think of a better idea.

Reply:
Another thought that comes to mind is that in the plugins you can report the version of the dependency as a metadata for the fu device when the devices are created.  By doing it with this approach you can even have the requirements checking in fu-engine just remove the updatable flag in this device in the instance that the appstream requirement is newer than the metadata for the fu device reported.

Reply:
That seems a bit magic to me.

Reply:
The part that I think is fragile is that metadata can be refreshed both before or after the device is made.  As long as the FuDevice has the library metadata in place when it's added, the daemon should be able to call a function to cross reference existing appstream metadata at that time (and remove flag if necessary).  If appstream metadata refreshes later that same that same function can be called at that time.